priam/src/main/java/com/netflix/priam/backup/SnapshotBackup.java

```diff
@@ -99,6 +103,21 @@ public void execute() throws Exception {
             sleeper.sleep(WAIT_TIME_MS);
         }
 
+        // Do not allow more than one snapshot to run at the same time. This is possible as this
+        // happens on CRON.
+        if (!lock.tryLock()) {
```

Comment:
You could execute multiple calls to Snapshot and since this is singleton "snapshotName" (local variable) changes in the middle. The previous snapshot will then never be processed. Ideally, we should not be executing another snapshot if the previous one is still RUNNING.